src/psyclone/transformations.py

```diff
@@ -2498,11 +2503,12 @@ def apply(self, node, options=None):
         if not options:
             options = {}
         default_present = options.get("default_present", False)
+        async_queue = options.get("async_queue", False)
 
         # Create a directive containing the nodes in node_list and insert it.
         directive = ACCKernelsDirective(
             parent=parent, children=[node.detach() for node in node_list],
-            default_present=default_present)
+            default_present=default_present, async_queue=async_queue)
```

Comment:
Please document this in the docstring and make it clear that it needs to be the same value as supplied to any preceding enter data directive.